x-pack/plugins/ingest_manager/server/services/setup.ts

```diff
@@ -157,18 +132,52 @@ function generateRandomPassword() {
   return Buffer.from(uuid.v4()).toString('base64');
 }
 
+async function addPackagesToConfig(
+  installedPackages: Installation[],
+  soClient: SavedObjectsClientContract,
+  config: AgentConfig,
+  defaultOutput: Output
+): Promise<void[]> {
+  const configWithDatasource = await agentConfigService.get(soClient, config.id, true);
+  if (!configWithDatasource) {
+    throw new Error('Config not found');
+  }
+  if (
+    configWithDatasource.datasources.length &&
+    typeof configWithDatasource.datasources[0] === 'string'
+  ) {
+    throw new Error('Config not found');
+  }
+  const addPackageToConfigPromises = installedPackages.map(async (installedPackage) => {
+    const packageShouldBeInstalled = DEFAULT_AGENT_CONFIGS_PACKAGES.some(
+      (packageName) => installedPackage.name === packageName
+    );
+    if (!packageShouldBeInstalled) {
+      return;
+    }
+
+    const isInstalled = configWithDatasource.datasources.some((d: Datasource | string) => {
+      return typeof d !== 'string' && d.package?.name === installedPackage.name;
+    });
+
+    if (!isInstalled) {
+      return addPackageToConfig(soClient, installedPackage, configWithDatasource, defaultOutput);
+    }
+
+  })
+  return Promise.all(addPackageToConfigPromises);
```

Comment:
Because of our datamodel, datasources is an array in the agent config, I do not think it's safe to call `addPackageToConfig` in parallel, this function is potentially not going to work 

https://github.com/elastic/kibana/blob/master/x-pack/plugins/ingest_manager/server/services/agent_config.ts/#L197-L220

Do you know if we can use script update in saved objects? I think this could make that function safer